### ISSUE TITLE
tf.math.sqrt returns infinite gradient for minimum positive float64 input although the analytic derivative is finite

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#124836]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

Python 3.13

### Current behavior?

`tf.math.sqrt` returns an incorrect gradient for the minimum positive float64 input.

For `sqrt(x)`, the derivative is `1 / (2 * sqrt(x))`. At `x = 5e-324`, the analytic high-precision reference is approximately `2.2494568972715982e+161`, which is a finite float64 value.

However, TensorFlow autodiff returns `inf` for this gradient. Peer autodiff backends such as Paddle and PyTorch compute the expected finite value, so this appears to be a backward-pass issue rather than a float64 limitation.

Expected behavior?

The gradient of `tf.math.sqrt` at `x = 5e-324` should be close to the analytic reference value `2.2494568972715982e+161`, not `inf`.

### Standalone code to reproduce the issue

```shell
import os
os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

import tensorflow as tf

x = 5e-324
truth = 2.2494568972715982e161
rtol = 1e-6

v = tf.Variable(x, dtype=tf.float64)

with tf.GradientTape() as tape:
    y = tf.math.sqrt(v)

g = tape.gradient(y, v)

actual = float(g)
rel_err = abs(actual - truth) / abs(truth)

print(f"x: {x!r}")
print(f"tensorflow gradient: {actual!r}")
print(f"expected gradient: {truth!r}")
print(f"relative error: {rel_err}")

if rel_err > rtol:
    print("BUG REPRODUCED: tf.math.sqrt gradient is infinite although the analytic float64 result is finite")
else:
    print("not reproduced")
```

### Relevant log output

```shell
x: 5e-324
tensorflow gradient: inf
expected gradient: 2.2494568972715982e+161
relative error: inf
BUG REPRODUCED: tf.math.sqrt gradient is infinite although the analytic float64 result is finite
```

**Comment by GodlyDonuts** [comment]
I reproduced this on CPU and traced it to the backward path reusing the flushed-to-zero sqrt output. I can work on a focused fix that recovers the finite float64 derivative for the entire positive-subnormal range, with eager/graph regression coverage.

**Comment by Venkat6871** [comment]
Hi @ALinrunrun ,
Thank you for raising your concern here.
A fix for this issue has been proposed in this [PR](https://github.com/tensorflow/tensorflow/pull/125178). Once it is merged, this issue will be resolved.

Hi @GodlyDonuts  ,
Thank you for raising the PR to address this issue. We truly appreciate your initiative and the effort in contributing the fix.
